File: mobile/src/main/java/uk/co/sentinelweb/bitwatcher/common/ui/picker/PickerPresenter.java
package uk.co.sentinelweb.bitwatcher.common.ui.picker;

import android.support.annotation.NonNull;


import java.util.List;

public class PickerPresenter<T> implements PickerContract.Presenter<T> {

    private final @NonNull
    PickerContract.View<T> view;

    private int selectedPosition;
    private List<PickerItemModel<T>> items;

    public PickerPresenter(@NonNull PickerContract.View<T> view) {
        this.view = view;
    }

    @Override
    public void bindData(@NonNull List<PickerItemModel<T>> items) {
        this.items = items;
        view.setData(items);
    }

    @Override
    public void setInitialPosition(int position) {
        selectedPosition = position;
        view.setInitialPosition(selectedPosition);
    }

    @Override
    public void updatePositionFromScroll(int position) {
        selectedPosition = position;
    }

    @Override
    public @NonNull
    PickerItemModel<T> getSelected() {
        return items.get(selectedPosition);
    }
}

File: mobile/src/main/java/uk/co/sentinelweb/bitwatcher/common/ui/picker/PickerPresenter.kt
package uk.co.sentinelweb.bitwatcher.common.ui.picker

class PickerPresenter<T>(private val view: PickerContract.View<T>) : PickerContract.Presenter<T> {

    private var selectedPosition: Int = 0
    private var items: List<PickerItemModel<T>>? = null

    override val selected: PickerItemModel<T>
        get() = items!![selectedPosition]

    override fun bindData(items: List<PickerItemModel<T>>) {
        this.items = items
        view.setData(items)
    }

    override fun setInitialPosition(position: Int) {
        selectedPosition = position
        view.setInitialPosition(selectedPosition)
    }

    override fun updatePositionFromScroll(position: Int) {
        selectedPosition = position
    }
}
